.eslintrc.js

```diff
@@ -255,8 +259,14 @@ module.exports = {
                 'plugin:jsx-a11y/recommended'
             ],
             rules: {
+                // Use TypeScript equivalent rules when required
+                'no-shadow': ['off'],
+                '@typescript-eslint/no-shadow': ['error'],
                 'no-useless-constructor': ['off'],
-                '@typescript-eslint/no-useless-constructor': ['error']
+                '@typescript-eslint/no-useless-constructor': ['error'],
+
+                'max-params': ['error', 7],
+                'sonarjs/cognitive-complexity': ['warn']
```

Comment:
I changed this to only warn on issues in ts(x) files to reduce the noise and try to highlight any complexity that is added in _new_ code.